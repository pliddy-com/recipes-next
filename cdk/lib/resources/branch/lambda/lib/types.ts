// TODO: share this defintion with client without crossing workspaces (shared space?)

export interface IRecipeSection {
  sectionTitle: string | null;
  sectionItems: (string | null)[];
}

export interface IRecipeChangeSet {
  abstract: string;
  cookTime: string | number;
  equipment: (string | null)[];
  id: string;
  ingredientsList: IRecipeSection[];
  instructionsList: IRecipeSection[];
  keywords: (string | null)[];
  notes: (string | null)[];
  prepTime: string | number;
  recipeYield: string | number;
  slug: string;
  title: string;
}
