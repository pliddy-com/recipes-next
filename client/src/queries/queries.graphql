# query for Taxonomy with Category & Subcategory levels
# getNavTaxonomy

query navMenuData {
  categories: taxonomyCollection(
    where: { slug: "categories" }
    order: title_ASC
    limit: 1
  ) {
    ...NavTaxonomy
  }

  cuisine: taxonomyCollection(
    where: { slug: "cuisine" }
    order: title_ASC
    limit: 1
  ) {
    ...NavTaxonomy
  }

  tags: tagCollection(order: title_ASC) {
    ...TagTaxonomy
  }
}

# Query for recipe slugs
# getRecipeSlugs

query recipeSlugs($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    __typename
    items {
      ...RecipeSlug
    }
  }
}

# query for category slugs
# getTagSlugs

query tagSlugs($where: TagFilter) {
  tagCollection(where: $where, order: title_ASC) {
    ...TagTaxonomy
  }
}

# query for lists of linked entries for tags and images
# getLinkedEntries

query linkedEntries {
  tags: tagCollection(order: title_ASC) {
    items {
      ...TagDefault
    }
  }
  images: assetCollection(order: fileName_ASC) {
    items {
      ...ImageDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)
# getRecipeList

query recipeList($where: TagFilter) {
  __typename
  tagCollection(where: $where, order: title_ASC, limit: 1) {
    __typename
    total
    items {
      ...ListPageItem
    }
  }
}

# Query for Recipe Collection for home (index) page
# getRecipeIndex

query recipeIndex($where: RecipeFilter) {
  __typename
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page
# getRecipePage

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    __typename
    items {
      ...RecipeDefault
    }
  }

  categoriesTaxonomy: taxonomyCollection(
    where: { slug: "categories" }
    order: title_ASC
    limit: 1
  ) {
    ...SchemaTaxonomy
  }

  cuisineTaxonomy: taxonomyCollection(
    where: { slug: "cuisine" }
    order: title_ASC
    limit: 1
  ) {
    ...SchemaTaxonomy
  }
}

query recipeEntryQuery($id: String!) {
  recipe(id: $id) {
    sys {
      id
      publishedVersion
    }

    ingredientsList
    instructionsList
  }
}

# used for recipe content structure migration
query recipeCollectionQuery {
  recipeCollection {
    total
    items {
      sys {
        id
        publishedVersion
      }

      ingredientsList
      instructionsList
    }
  }
}
