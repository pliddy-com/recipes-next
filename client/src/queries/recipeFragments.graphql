# Fragment with slug for Recipe

fragment RecipeSlug on Recipe {
  __typename
  slug
}

# Fragments with default fields for Recipe-specific sections:
#   * Ingredients
#   * Instructions
#   * Image
#   * RecipeDefault (full)
#   * RecipeSummary (card)

fragment IngredientsDefault on IngredientSection {
  __typename
  title
  slug
  label
  ingredientList
}

fragment InstructionsDefault on InstructionSection {
  __typename
  title
  slug
  label
  instructionList
}

fragment ImageDefault on Asset {
  __typename
  title
  description
  contentType
  fileName
  size
  url
  height
  width
}

fragment DescriptionDefault on Recipe {
  description {
    json
    links {
      assets {
        block {
          sys {
            id
          }
          __typename
          title
          url
        }
        hyperlink {
          sys {
            id
          }
          __typename
          title
          url
        }
      }
      entries {
        inline {
          __typename
          sys {
            id
          }
          ... on Recipe {
            slug
            title
          }
        }
        hyperlink {
          __typename
          sys {
            id
          }
        }
      }
    }
  }
}

fragment RecipeDefault on Recipe {
  __typename
  title
  slug
  sys {
    firstPublishedAt
  }
  ...DescriptionDefault
  abstract
  image {
    ...ImageDefault
  }
  recipeYield
  prepTime
  cookTime
  ingredientsCollection(limit: 6) {
    items {
      ...IngredientsDefault
    }
  }
  equipment
  instructionsCollection(limit: 6) {
    items {
      ...InstructionsDefault
    }
  }
  notes
  tagsCollection(limit: 10) {
    items {
      ...TagDefault
    }
  }
  keywords
}

fragment RecipeSummary on Recipe {
  __typename
  sys {
    firstPublishedAt
  }
  title
  slug
  abstract
  image {
    ...ImageDefault
  }
  tagsCollection(limit: 6) {
    items {
      ...TagDefault
    }
  }
}
