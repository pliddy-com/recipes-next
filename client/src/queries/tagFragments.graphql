## Fragments with default fields for:
#   * Tag

fragment TagDefault on Tag {
  __typename
  title
  slug
}

## Fragments with default fields for:
#   * Tag

fragment TagEntry on Tag {
  sys {
    id
  }
  __typename
  title
  slug
}

# Fragment with fields for Tag with linkedFrom Recipes

fragment TagWithLinks on Tag {
  __typename
  title
  slug

  linkedFrom {
    recipeCollection(limit: 1) {
      total
    }
  }
}

# fragment for items on list page queries
# TODO: investigate reducing complexity
fragment ListPageItem on Tag {
  __typename
  slug
  title
  linkedFrom {
    recipeCollection(limit: 13) {
      total
      items {
        ...RecipeSummary
      }
    }
  }
}
