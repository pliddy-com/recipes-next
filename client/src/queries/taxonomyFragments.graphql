# Fragments with default fields for:
#   * Taxonomy

fragment TaxonomyDefault on Taxonomy {
  __typename
  title
  slug
}

# Fragment with Fields for Taxonomy with no children

fragment TaxonomyRoot on Taxonomy {
  # ...TaxonomyDefault
  __typename
  title
  slug

  tag {
    sys {
      id
    }
    __typename
    title
    slug
  }
}

# Fragment with Fields for Taxonomy with children

fragment TaxonomyWithChildren on Taxonomy {
  # ...TaxonomyDefault
  __typename
  title
  slug

  tag {
    ...TagWithLinks
  }
  childrenCollection {
    total
    items {
      ...TagWithLinks
      ...TaxonomyRoot
    }
  }
}

# Fragment with Fields for Taxonomy for Nav query

fragment NavTaxonomy on TaxonomyCollection {
  __typename
  total
  items {
    ...TaxonomyDefault

    tag {
      __typename
      title
      slug
    }

    childrenCollection(limit: 26) {
      total
      items {
        ... on Tag {
          __typename
          title
          slug

          linkedFrom {
            recipeCollection(limit: 1) {
              total
            }
          }
        }
        ...TaxonomyWithChildren
      }
    }
  }
}

# Fragment with Fields for Tag used with Taxonomy queries

fragment TagTaxonomy on TagCollection {
  __typename
  total
  items {
    ...TagDefault
    linkedFrom {
      recipeCollection(limit: 1) {
        total
      }
    }
  }
}

# Fragment with Fields for Taxonomy used with schema queries

fragment SchemaTaxonomy on TaxonomyCollection {
  __typename
  items {
    childrenCollection(limit: 24) {
      total
      items {
        ... on Tag {
          __typename
          title
          slug
        }
        ... on Taxonomy {
          ...TaxonomyDefault
        }
      }
    }
  }
}
