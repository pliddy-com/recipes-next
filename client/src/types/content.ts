export interface IRecipeSection {
  sectionTitle: string | null;
  sectionItems: string[] | null;
}

export interface IRecipeChangeSet {
  abstract: string;
  cookTime: string | number;
  id: string;
  prepTime: string | number;
  recipeYield: string | number;
  slug: string;
  title: string;
}
