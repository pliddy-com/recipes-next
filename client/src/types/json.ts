export interface IRecipeSection {
  sectionTitle: string | null;
  sectionItems: string[] | null;
}
