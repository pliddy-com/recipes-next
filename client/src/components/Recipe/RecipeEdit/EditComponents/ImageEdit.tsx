import { useEffect, useState } from 'react';

import Box from '@mui/material/Box';
import ImageList from '@mui/material/ImageList';
import ImageListItem from '@mui/material/ImageListItem';
import ImageListItemBar from '@mui/material/ImageListItemBar';
import Stack from '@mui/material/Stack';

import AddCircleIcon from '@mui/icons-material/AddCircle';
import CloudUploadIcon from '@mui/icons-material/CloudUpload';
import IconButton from '@mui/material/IconButton';
import ImageIcon from '@mui/icons-material/Image';

import DynamicImage from 'components/Image/DynamicImage/DynamicImage';
import NavIconButton from 'components/Navigation/Buttons/NavIconButton/NavIconButton';

import { Breakpoints } from 'lib/responsiveImage';

import { AspectRatio } from 'theme/values/images';

import { ImageDefaultFragment } from 'types/queries';

export interface IImageEdit {
  aspectRatio?: AspectRatio;
  breakpoints: Breakpoints[];
  image: ImageDefaultFragment;
  imageList: (ImageDefaultFragment | null)[] | undefined;
  onChange: ({ value }: { value: ImageDefaultFragment }) => void;
  preload?: boolean;
  thumbBreakpoints: Breakpoints[];
}

const ImageEdit = ({
  aspectRatio,
  image,
  imageList,
  breakpoints,
  onChange,
  preload,
  thumbBreakpoints
}: IImageEdit) => {
  const [selectedImage, setSelectedImage] =
    useState<ImageDefaultFragment>(image);

  const [openChangeImage, setOpenChangeImage] = useState<boolean>(false);
  const [openUploadImage, setOpenUploadImage] = useState<boolean>(false);

  useEffect(() => {
    setSelectedImage(image);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleClick = (image: ImageDefaultFragment) => {
    setSelectedImage(image);
    onChange({ value: image });
    setOpenChangeImage(false);
  };

  const showChangeImage = () => {
    setOpenChangeImage(!openChangeImage);
    setOpenUploadImage(false);
  };

  const showUploadImage = () => {
    setOpenUploadImage(!openUploadImage);
    setOpenChangeImage(false);
  };

  return imageList ? (
    <Stack className="image-edit">
      <Box className="image original">
        <DynamicImage
          aspectRatio={aspectRatio}
          breakpoints={breakpoints}
          image={selectedImage}
          preload={preload}
        />
      </Box>

      <Stack
        direction={'row'}
        sx={{ justifyContent: 'space-between', marginBottom: '1.5rem' }}
      >
        <NavIconButton
          ariaLabel={'change image button'}
          className="edit-button"
          hideLabel={false}
          icon={<ImageIcon />}
          isMenu={true}
          isOpen={openChangeImage}
          label={'Change Image'}
          onClick={showChangeImage}
          variant="outlined"
        />

        <NavIconButton
          ariaLabel={'upload image button'}
          className="edit-button"
          disabled={openUploadImage}
          hideLabel={false}
          icon={<CloudUploadIcon />}
          label={'Add Image'}
          onClick={showUploadImage}
          variant="outlined"
        />
      </Stack>

      {openChangeImage && (
        <ImageList className="edit-image-list">
          {imageList.map((image) =>
            image ? (
              <ImageListItem key={image.url}>
                <Box className="image">
                  <DynamicImage
                    aspectRatio={aspectRatio}
                    breakpoints={thumbBreakpoints}
                    image={image}
                    preload={preload}
                  />
                </Box>
                <ImageListItemBar
                  actionIcon={
                    <IconButton
                      aria-label={`select image ${image.fileName}`}
                      onClick={() => handleClick(image)}
                    >
                      <AddCircleIcon />
                    </IconButton>
                  }
                  title={image.fileName}
                />
              </ImageListItem>
            ) : null
          )}
        </ImageList>
      )}
    </Stack>
  ) : null;
};

export default ImageEdit;
