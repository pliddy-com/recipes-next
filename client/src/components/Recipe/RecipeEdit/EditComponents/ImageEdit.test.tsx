import '@testing-library/jest-dom';

// import testing-library methods
import { fireEvent, render, waitFor } from '@testing-library/react';
import ImageEdit, { IImageEdit } from './ImageEdit';

import { ImageDefaultFragment } from 'types/queries';

describe('ImageEdit', () => {
  const aspectRatio = '4 / 3';

  const breakpoints = [
    {
      viewMin: 1000,
      imgWidth: 400
    },
    {
      viewMin: 500,
      imgWidth: 200
    },
    {
      imgWidth: 100
    }
  ];

  const image: ImageDefaultFragment = {
    __typename: 'Asset',
    sys: {
      id: 'img-id'
    },
    description: 'Image Description',
    fileName: 'image-file.jpg',
    height: 300,
    title: 'Image Title',
    url: 'https://image.url',
    width: 400
  };

  const imageList: ImageDefaultFragment[] = [
    {
      __typename: 'Asset',
      sys: {
        id: 'img-id-1'
      },
      description: 'Image Description 1',
      fileName: 'image-file-1.jpg',
      height: 300,
      title: 'Image Title 1',
      url: 'https://image.url/1',
      width: 400
    },
    {
      __typename: 'Asset',
      sys: {
        id: 'img-id-2'
      },
      description: 'Image Description 2',
      height: 300,
      fileName: 'image-file-2.jpg',
      title: 'Image Title 2',
      url: 'https://image.url/2',
      width: 400
    }
  ];

  describe('when there is content', () => {
    it('renders the component', async () => {
      const props: IImageEdit = {
        aspectRatio,
        breakpoints,
        image,
        imageList,
        onChange: jest.fn()
      };

      const { asFragment, getByRole } = render(<ImageEdit {...props} />);

      await waitFor(async () => {
        expect(asFragment()).toMatchSnapshot();
      });

      const clickImage = getByRole('button', {
        name: 'select image image-file-1.jpg'
      });

      clickImage && fireEvent.click(clickImage);
    });
  });
  describe('when imageList is undefined', () => {
    it('it does not render the component', async () => {
      // const tags = undefined as unknown as TagDefaultFragment[];
      const props: IImageEdit = {
        aspectRatio,
        breakpoints,
        image,
        imageList: undefined,
        onChange: jest.fn()
      };

      render(<ImageEdit {...props} />);

      const { asFragment } = render(<ImageEdit {...props} />);

      await waitFor(async () => {
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('when images in list imageList are null', () => {
    it('it does not render the component', async () => {
      // const tags = undefined as unknown as TagDefaultFragment[];
      const props: IImageEdit = {
        aspectRatio,
        breakpoints,
        image,
        imageList: [null],
        onChange: jest.fn()
      };

      render(<ImageEdit {...props} />);

      const { asFragment } = render(<ImageEdit {...props} />);

      await waitFor(async () => {
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
