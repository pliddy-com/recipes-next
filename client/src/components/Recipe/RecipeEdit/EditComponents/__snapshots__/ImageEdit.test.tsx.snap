// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageEdit when imageList is undefined it does not render the component 1`] = `<DocumentFragment />`;

exports[`ImageEdit when images in list imageList are null it does not render the component 1`] = `
<DocumentFragment>
  <div
    class="image-edit css-nen11g-MuiStack-root"
  >
    <div
      class="image original MuiBox-root css-0"
    >
      <picture>
        <source
          media="(min-width: 1000px)"
          sizes="400px"
          srcset="https://image.url?w=400&h=300&fit=fill&fm=webp&q=75 1x, https://image.url?w=800&h=600&fit=fill&fm=webp&q=75 2x"
        />
        <source
          media="(min-width: 500px) and (max-width: 999px)"
          sizes="200px"
          srcset="https://image.url?w=200&h=150&fit=fill&fm=webp&q=75 1x, https://image.url?w=400&h=300&fit=fill&fm=webp&q=75 2x"
        />
        <source
          media="(max-width: 499px)"
          sizes="100px"
          srcset="https://image.url?w=100&h=75&fit=fill&fm=webp&q=75 1x, https://image.url?w=200&h=150&fit=fill&fm=webp&q=75 2x"
        />
        <img
          alt="Image Description"
          class="dynamicImage"
          fetchpriority="auto"
          height="300"
          loading="lazy"
          src="https://image.url?w=400&h=300&fit=fill&q=75"
          width="400"
        />
      </picture>
    </div>
    <ul
      class="MuiImageList-root MuiImageList-standard css-193qa7j-MuiImageList-root"
      style="grid-template-columns: repeat(3, 1fr); gap: 4px;"
    />
  </div>
</DocumentFragment>
`;

exports[`ImageEdit when there is content renders the component 1`] = `
<DocumentFragment>
  <div
    class="image-edit css-nen11g-MuiStack-root"
  >
    <div
      class="image original MuiBox-root css-0"
    >
      <picture>
        <source
          media="(min-width: 1000px)"
          sizes="400px"
          srcset="https://image.url?w=400&h=300&fit=fill&fm=webp&q=75 1x, https://image.url?w=800&h=600&fit=fill&fm=webp&q=75 2x"
        />
        <source
          media="(min-width: 500px) and (max-width: 999px)"
          sizes="200px"
          srcset="https://image.url?w=200&h=150&fit=fill&fm=webp&q=75 1x, https://image.url?w=400&h=300&fit=fill&fm=webp&q=75 2x"
        />
        <source
          media="(max-width: 499px)"
          sizes="100px"
          srcset="https://image.url?w=100&h=75&fit=fill&fm=webp&q=75 1x, https://image.url?w=200&h=150&fit=fill&fm=webp&q=75 2x"
        />
        <img
          alt="Image Description"
          class="dynamicImage"
          fetchpriority="auto"
          height="300"
          loading="lazy"
          src="https://image.url?w=400&h=300&fit=fill&q=75"
          width="400"
        />
      </picture>
    </div>
    <ul
      class="MuiImageList-root MuiImageList-standard css-193qa7j-MuiImageList-root"
      style="grid-template-columns: repeat(3, 1fr); gap: 4px;"
    >
      <li
        class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <div
          class="image MuiBox-root css-0"
        >
          <picture>
            <source
              media="(min-width: 960px)"
              sizes="300px"
              srcset="https://image.url/1?w=300&h=225&fit=fill&fm=webp&q=75 1x, https://image.url/1?w=600&h=450&fit=fill&fm=webp&q=75 2x"
            />
            <source
              media="(min-width: 656px) and (max-width: 959px)"
              sizes="200px"
              srcset="https://image.url/1?w=200&h=150&fit=fill&fm=webp&q=75 1x, https://image.url/1?w=400&h=300&fit=fill&fm=webp&q=75 2x"
            />
            <source
              media="(max-width: 655px)"
              sizes="200px"
              srcset="https://image.url/1?w=200&h=150&fit=fill&fm=webp&q=75 1x, https://image.url/1?w=400&h=300&fit=fill&fm=webp&q=75 2x"
            />
            <img
              alt="Image Description 1"
              class="dynamicImage"
              fetchpriority="auto"
              height="225"
              loading="lazy"
              src="https://image.url/1?w=300&h=225&fit=fill&q=75"
              width="300"
            />
          </picture>
        </div>
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom css-ztrqiq-MuiImageListItemBar-root"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-14q7c0c-MuiImageListItemBar-titleWrap"
          >
            <div
              class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
            >
              image-file-1.jpg
            </div>
          </div>
          <div
            class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-mjbero-MuiImageListItemBar-actionIcon"
          >
            <button
              aria-label="select image image-file-1.jpg"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yon2v9-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </li>
      <li
        class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <div
          class="image MuiBox-root css-0"
        >
          <picture>
            <source
              media="(min-width: 960px)"
              sizes="300px"
              srcset="https://image.url/2?w=300&h=225&fit=fill&fm=webp&q=75 1x, https://image.url/2?w=600&h=450&fit=fill&fm=webp&q=75 2x"
            />
            <source
              media="(min-width: 656px) and (max-width: 959px)"
              sizes="200px"
              srcset="https://image.url/2?w=200&h=150&fit=fill&fm=webp&q=75 1x, https://image.url/2?w=400&h=300&fit=fill&fm=webp&q=75 2x"
            />
            <source
              media="(max-width: 655px)"
              sizes="200px"
              srcset="https://image.url/2?w=200&h=150&fit=fill&fm=webp&q=75 1x, https://image.url/2?w=400&h=300&fit=fill&fm=webp&q=75 2x"
            />
            <img
              alt="Image Description 2"
              class="dynamicImage"
              fetchpriority="auto"
              height="225"
              loading="lazy"
              src="https://image.url/2?w=300&h=225&fit=fill&q=75"
              width="300"
            />
          </picture>
        </div>
        <div
          class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom css-ztrqiq-MuiImageListItemBar-root"
        >
          <div
            class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-14q7c0c-MuiImageListItemBar-titleWrap"
          >
            <div
              class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
            >
              image-file-2.jpg
            </div>
          </div>
          <div
            class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-mjbero-MuiImageListItemBar-actionIcon"
          >
            <button
              aria-label="select image image-file-2.jpg"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yon2v9-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
